system/jlib/jsocket.cpp

```diff
@@ -2125,15 +2125,27 @@ bool CSocket::check_connection()
         rc = sendto(sock, NULL, 0, 0, &u.sa, ul);
     }
     else {
+#ifdef _WIN32
         rc = send(sock, NULL, 0, SEND_FLAGS);
+
+#else
+        char buffer;
+        rc = recv(sock, &buffer, sizeof(buffer), MSG_PEEK | MSG_DONTWAIT);
```

Comment:
I might have done a poll() with a 0 timeout and if it returned ready for read then read() with MSG_PEEK but not MSG_DONTWAIT and check for <= 0.
But not sure if that would be any different than here, except for not having to test for EWOULDBLOCK/EAGAIN.